### PR TITLE
Implement `ClusterAdapter` for ES7.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This change is implementing a `ClusterAdapter` for ES7. It provides cluster-related functionality and is consumed mainly by the `Cluster` class.
    
During the implementation of this using the Elasticsearch high-level REST client, it became obvious that it is not supporting API endpoints that Jest supported, namely the `_cat` and the `_nodes` endpoints. For the `_cat` API, a `CatApi` class was implemented, that returns proper value objects for the responses. For the other missing APIs it was different, as only a small number of keys were used from the responses, with little overlap. Therefore a `PlainJsonApi` class created, that allows lightweight querying of single endpoints, returning the plain JSON structure.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/ClusterAdapterES6.java [hunk 1]
```diff
@@ -74,8 +73,7 @@ private HealthStatus extractHealthStatus(JsonNode node) {
 
     @Override
```
[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/ClusterAdapterES6.java:74]: :heavy_check_mark:

File: graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/ClusterAdapterES6.java [hunk 2]
```diff
@@ -74,8 +73,7 @@ private HealthStatus extractHealthStatus(JsonNode node) {
```
[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/ClusterAdapterES6.java]: :heavy_check_mark:

File: graylog-storage-elasticsearch6/src/test/java/org/graylog/storage/elasticsearch6/ClusterES6IT.java [hunk 3]
```diff
@@ -18,6 +21,18 @@
     @Rule
     public final ElasticsearchInstance elasticsearch = ElasticsearchInstanceES6.create();
 
+    @Test
+    public void getClusterAllocationDiskSettings() {
+        final ClusterAllocationDiskSettings clusterAllocationDiskSettings = cluster.getClusterAllocationDiskSettings();
+
+        //Default Elasticsearch settings in Elasticsearch 5.6
```
[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog-storage-elasticsearch6/src/test/java/org/graylog/storage/elasticsearch6/ClusterES6IT.java]: :heavy_check_mark:

File: graylog-storage-elasticsearch6/src/test/java/org/graylog/storage/elasticsearch6/ClusterES6IT.java [hunk 4]
```diff
@@ -18,6 +21,18 @@
...
+        assertThat(clusterAllocationDiskSettings.ThresholdEnabled()).isTrue();
+        assertThat(clusterAllocationDiskSettings.watermarkSettings().type()).isEqualTo(WatermarkSettings.SettingsType.PERCENTAGE);
+        assertThat(clusterAllocationDiskSettings.watermarkSettings().low()).isEqualTo(85D);
+        assertThat(clusterAllocationDiskSettings.watermarkSettings().high()).isEqualTo(90D);
+        assertThat(clusterAllocationDiskSettings.watermarkSettings().floodStage()).isNull();
```
[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog-storage-elasticsearch6/src/test/java/org/graylog/storage/elasticsearch6/ClusterES6IT.java]: :heavy_check_mark:

File: graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ClusterAdapterES7.java [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+                        response.getRelocatingShards(),
+                        response.getActivePrimaryShards(),
+                        response.getInitializingShards(),
+                        response.getUnassignedShards(),
+                        response.isTimedOut()
+                ))
+                .orElseThrow(() -> new ElasticsearchException("Unable to retrieve shard stats."));
+    }
+
+    private Optional<ClusterHealthResponse> clusterHealth(Collection<? extends String> indices) {
```
[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ClusterAdapterES7.java]: :heavy_check_mark:

File: graylog2-server/src/main/java/org/graylog2/system/stats/ClusterStatsService.java [hunk 6]
```diff
@@ -110,7 +113,25 @@ private long countDashboards() {
 
     public ElasticsearchStats elasticsearchStats() {
```
[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog2-server/src/main/java/org/graylog2/system/stats/ClusterStatsService.java:111]: :heavy_check_mark:

File: graylog2-server/src/main/java/org/graylog2/system/stats/ClusterStatsService.java [hunk 7]
```diff
@@ -53,8 +51,8 @@
     private final AlertService alertService;
     private final AlarmCallbackConfigurationService alarmCallbackConfigurationService;
     private final DashboardService dashboardService;
+    private final Cluster cluster;
     private final IndexSetRegistry indexSetRegistry;
```
Comment (alex-konn) [inline graylog2-server/src/main/java/org/graylog2/system/stats/ClusterStatsService.java:55]: After extracting `elasticsearchStats` to `Cluster` we can now get rid of `IndexSetRegistry` here.

Comment (dennisoelkers) [reply graylog2-server/src/main/java/org/graylog2/system/stats/ClusterStatsService.java]: :heavy_check_mark:

**Pull request merged**